packages/asset-swapper/src/utils/alt_mm_implementation_utils.ts

```diff
@@ -0,0 +1,248 @@
+                );
+                makerAmount = Web3Wrapper.toBaseUnitAmount(
+                    new BigNumber(altIndicativeQuoteReponse.amount)
+                        .times(new BigNumber(altIndicativeQuoteReponse.price))
+                        .decimalPlaces(altPair.quoteAssetDecimals, BigNumber.ROUND_DOWN),
+                    altPair.quoteAssetDecimals,
+                );
+            } else if (altIndicativeQuoteReponse.value) {
+                // base asset = taker asset, quote asset = maker asset, value specified
+                makerAmount = Web3Wrapper.toBaseUnitAmount(
+                    new BigNumber(altIndicativeQuoteReponse.value),
+                    altPair.quoteAssetDecimals,
+                );
+                takerAmount = Web3Wrapper.toBaseUnitAmount(
+                    new BigNumber(altIndicativeQuoteReponse.value)
+                        .dividedBy(new BigNumber(altIndicativeQuoteReponse.price))
+                        .decimalPlaces(altPair.baseAssetDecimals, BigNumber.ROUND_DOWN),
+                    altPair.baseAssetDecimals,
+                ).integerValue(BigNumber.ROUND_DOWN);
+            } else {
+                throw new Error('neither amount or value were specified');
+            }
+        } else {
+            throw new Error(`Base, quote tokens don't align with maker, taker tokens`);
+        }
+    }
+
+    return {
+        makerToken,
+        makerAmount,
+        takerToken,
+        takerAmount,
+        // HACK: alt implementation does not return an expiration with indicative quotes
+        // return now + { IMPUTED EXPIRY SECONDS } to have it included after order checks
+        // tslint:disable-next-line:custom-no-magic-numbers
+        expiry: new BigNumber(
+            new BigNumber(Date.now() / 1000).integerValue(BigNumber.ROUND_DOWN).plus(IMPUTED_EXPIRY_SECONDS),
+        ),
+    };
+}
+
+export async function returnQuoteFromAltMMAsync<ResponseT>(
+    url: string,
+    apiKey: string,
+    profile: string,
+    integratorKey: string,
+    quoteModel: AltQuoteModel,
+    makerToken: string,
+    takerToken: string,
+    maxResponseTimeMs: number,
+    altRfqtAssetOfferings: AltRfqtMakerAssetOfferings,
+    takerRequestQueryParams: TakerRequestQueryParams,
+    axiosInstance: AxiosInstance,
+): Promise<{ data: ResponseT; status: number }> {
+    const altPair = getAltMarketInfo(
+        altRfqtAssetOfferings[url],
+        takerRequestQueryParams.buyTokenAddress,
+        takerRequestQueryParams.sellTokenAddress,
+    );
+
+    if (altPair) {
+        const side =
+            altPair.baseAsset === takerRequestQueryParams.buyTokenAddress ? AltQuoteSide.Buy : AltQuoteSide.Sell;
+
+        // comparison price needs to be quote/base
+        // in the standard implementation, it's maker/taker
+        const altComparisonPrice =
+            altPair.quoteAsset === makerToken
+                ? takerRequestQueryParams.comparisonPrice
```

Comment:
Wouldn't `takerRequestQueryParams.comparisonPrice` be already `undefined`? do we need that tenary statement?

Reply:
don't understand the comment